### DISCUSSION TITLE
Realm

### DISCUSSION THREAD

**Discussion by ghost** [lit-iphone-course/lit-iphone-course#41] [Question-original] [answered]
(1) 実現したいこと
テーブルビューのセルを押すと、書いた内容が呼び出される
(2) 実現するために自分が試した内容とその結果
教科書など
homeviewcontrollerではできたのですが、editviewcontrollerではできません。初歩的な質問で申し訳ないです
(3) 調べた内容や資料
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
`import UIKit
import RealmSwift

class EditViewController: UIViewController, UITextFieldDelegate, UITextViewDelegate {
    
    let realm = try! Realm()
    
//    var todosInEdit = [Memo]()
    

    @IBOutlet var titleLabel: UILabel!
    @IBOutlet var contenLabel: UILabel!
    

    override func viewDidLoad() {
        super.viewDidLoad()
        
//        todosInEdit = Array(realm.objects(Memo.self))
        
        let memo: Memo? = edit()
        
        titleLabel.text = memo?.title
        contenLabel.text = memo?.content
     // Do any additional setup after loading the view.
        
   }
    
    //メモの情報を読み出す
    func edit() -> Memo? {
        return realm.objects(Memo.self).first
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
`
(5) GitHubのリンク
https://github.com/j-fuuuu/frog2.git

**Comment by txxxxc** [comment]
GitHubのリポジトリの最終更新が6時間前になってるので、一旦質問した時点のコードをpushしてもらえるとありがたいです！手元で確認したいので！

**Comment by txxxxc** [comment, marked as answer]
まずスタンスとして編集画面にでRealmからデータを引っ張ってくるのではなく、一覧画面でRealmから取得したデータを編集画面に渡してあげる流れのほうが一般的な実装です！
なので以下のような流れで実装してあげるときれいに実装できると思います！
1. セルがタップされた時にタップされたセルのindexをつかってRealmから取得したデータの中から一つを取り出す
2. それを編集画面に渡しながら画面遷移する


セルがタップされたときにイベントを発火させる方法についてはこちらを参考にしてもらえるとよさそうです！
[【Swift/iOS】UITableViewのセルをタップした時のアクションを実装する | カピ通信](https://capibara1969.com/634/)
一回試してみてくださいー！

ps. こっちのエラーは治りそうですか...?ヤバそうだったら教えて下さい！
[変更したimageviewとlabelのデータをアプリを再起動しても変更した情報が残っているようにしたい · Discussion #34 · lit-iphone-course/lit-iphone-course](https://github.com/lit-iphone-course/lit-iphone-course/discussions/34)

**Reply by ghost** [reply]
すみません。先ほど上げ直しました。

https://github.com/j-fuuuu/frog09231530.git




送っていただいた、URLの
` func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        // セルの選択を解除
        tableView.deselectRow(at: indexPath, animated: true)
        
        // 別の画面に遷移
        performSegue(withIdentifier: "editMemo", sender: nil)
    }`

をすると、セルを押すと2回画面が切り替わるようになったのですが、この場合このコードは入れる必要はなかったのでしょうか？


前回の質問ですが、labelの方はできているとおもますが、imageviewの方はうまくいっていないです。

**Reply by txxxxc** [reply]
あ、もしかして毎回新しいリポジトリ作ってる？？
GitHubのリポジトリは一つでOKで、その一つのリポジトリに変更履歴を記録していくのが一般的なやり方なので、覚えておいてもらえると良さそうです！

参考はこちらです！
https://zenn.dev/kueharx/articles/ebd14c46f02211
https://qiita.com/sakamotoyuya/items/fd6c30e672860ba89759

**Comment by txxxxc** [comment]
> をすると、セルを押すと2回画面が切り替わるようになったのですが、この場合このコードは入れる必要はなかったのでしょうか？

2回画面が切り替わるようになっている原因は画面遷移の繋ぎ方の問題です！この記事の`任意のタイミングでの画面遷移（Manual Segue）`の用な感じでsegueを繋ぎ直してみてくださいー！
[【Swift/iOS】Segueを使った画面遷移 | カピ通信](https://capibara1969.com/229/)

> 前回の質問ですが、labelの方はできているとおもますが、imageviewの方はうまくいっていないです。

なるほど！！そっちも一緒に直していきましょう！！この問題についてはもう一個の投稿の方でやり取りしましょうー！

**Reply by ghost** [reply]
segueの方は解決しました
githubの件了解です
